lib/pbench/test/unit/server/database/test_datasets_metadata_db.py

```diff
@@ -427,3 +427,105 @@ def test_mutable_server_bad(self, server_config, attach_dataset, value, message)
             Metadata.setvalue(ds, "server.deletion", value)
         assert str(exc.value) == message
         assert Metadata.getvalue(ds, "server.deletion") == deletion
+
+    def test_mutable_origin(self, server_config, attach_dataset):
+        """
+        Set the dataset origin origin metadata
+        """
+        ds = Dataset.query(name="drb")
+        Metadata.setvalue(ds, "server.origin", "RIYA")
+        assert Metadata.getvalue(ds, "server.origin") == "RIYA"
+
+    @pytest.mark.parametrize(
+        "value,message",
+        [
+            (
+                True,
+                "Metadata key 'server.origin' value True for dataset (3)|drb must be a string",
+            ),
+            (
+                [],
+                "Metadata key 'server.origin' value [] for dataset (3)|drb must be a string",
+            ),
+            (
+                1,
+                "Metadata key 'server.origin' value 1 for dataset (3)|drb must be a string",
+            ),
+        ],
+    )
+    def test_mutable_origin_bad(self, server_config, attach_dataset, value, message):
+        """
+        Try out some invalid deletion time values.
+
+        The value must be a valid date/time string, and it must be within the
+        server's maximum retention threshold from the dataset's upload time.
+        """
+        ds = Dataset.query(name="drb")
+        with pytest.raises(MetadataBadValue) as exc:
+            Metadata.setvalue(ds, "server.origin", value)
+        assert str(exc.value) == message
+        assert Metadata.getvalue(ds, "server.origin") is None
+
+    @pytest.mark.parametrize(
+        "value,result",
+        [
+            ("true", True),
+            ("false", False),
+            ("t", True),
+            ("f", False),
+            ("y", True),
+            ("yEs", True),
+            ("n", False),
+            ("No", False),
+            (1, True),
+            (0, False),
+            ("True", True),
+            ("FAlSe", False),
+        ],
```

Comment:
Odd that I overlooked the booleans. I think it's because, as I mentioned earlier, in the only context where `server.archiveonly` is currently meaningful (on upload) we can only specify string values anyway.

As for `int` values ... well the only "interesting" numeric values in this context are 0 and non-zero, and I decided to stop short of adding every possible integer to the list ...

Reply:
> I decided to stop short of adding every possible integer to the list ...

😆  Yes, but, I submit that just zero and one are not a representative sample of the full set.  😉 

> in the only context where `server.archiveonly` is currently meaningful (on upload) we can only specify string values anyway.

Then, I suggest that we support only string values and "string-y" notions of true and false (and not bother with `int`'s, `bool`'s, and other non-string types) -- this makes both the implementation and the tests simpler.  And, if a need arises for other types, we can implement that later.